File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-3.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode temp = new ListNode(-1);
        ListNode temp2 = temp;

        ListNode traverseList1 = list1;
        ListNode traverseList2 = list2;

        while(traverseList1!=null && traverseList2!=null){
            if(traverseList1.val <= traverseList2.val){
                temp2.next = traverseList1;
                traverseList1 = traverseList1.next;
            }
            else if(traverseList1.val > traverseList2.val){
                temp2.next = traverseList2;
                traverseList2 = traverseList2.next;
            }
            temp2 = temp2.next;
        }

        if(traverseList1!=null){
            temp2.next = traverseList1;
        }
        else{
            temp2.next = traverseList2;
        }

        return temp.next;
    }
}

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-3.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        if not lists:
            return None
        def merge(array, index):
            copy = array
            iterate = copy
            temp = ListNode(-1)
            second = temp
            copySecond = lists[index]
            iterate2 = copySecond
            while copy and copySecond:
                if copy.val <= copySecond.val:
                    second.next = copy
                    copy = copy.next
                else:
                    second.next = copySecond
                    copySecond = copySecond.next
                second = second.next
            if copy:
                second.next = copy
            if copySecond:
                second.next = copySecond
            return temp.next
        
        mergedArray = lists[0]
        ind = 1
        while ind < len(lists):
            currArray = merge(mergedArray, ind)
            mergedArray = currArray
            ind+=1
        return mergedArray